x-pack/plugins/apm/server/lib/services/get_service_metadata_details.ts

```diff
@@ -60,34 +61,33 @@ interface ServiceMetadataDetails {
 export async function getServiceMetadataDetails({
   serviceName,
   setup,
+  searchAggregatedTransactions,
 }: {
   serviceName: string;
   setup: Setup & SetupTimeRange;
+  searchAggregatedTransactions: boolean;
 }): Promise<ServiceMetadataDetails> {
   const { start, end, apmEventClient } = setup;
 
   const filter = [
     { term: { [SERVICE_NAME]: serviceName } },
-    { term: { [PROCESSOR_EVENT]: ProcessorEvent.transaction } },
     { range: rangeFilter(start, end) },
-    ...setup.esFilter,
-  ];
-
-  const should = [
-    { exists: { field: CONTAINER } },
-    { exists: { field: KUBERNETES } },
-    { exists: { field: CLOUD } },
-    { exists: { field: HOST } },
-    { exists: { field: AGENT } },
   ];
 
   const params = {
     apm: {
-      events: [ProcessorEvent.transaction],
+      events: [
+        getProcessorEventForAggregatedTransactions(
+          searchAggregatedTransactions
+        ),
+        ProcessorEvent.error,
+        ProcessorEvent.span,
```

Comment:
I think we can drop spans here (I know I suggested this 🙂). We have no tab for spans, so not querying those will not create edge-cases where the user has spans and is looking at the spans tab, but no errors, metrics or transactions.